### PR TITLE
Don't pass `-fvisibility=default` unless building with `-fPIC`. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This means that, by default, we will get `-fvisibility=hidden` since that is the default in clang when building for wasm.

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -386,7 +386,7 @@ def get_clang_flags(user_args):
     # enabled so that it can be added to the attributes in LLVM IR.
     if settings.SUPPORT_LONGJMP == 'wasm':
       flags.append('-mexception-handling')
-
+    #flags.append('-fwhole-program-vtables')
```
Comment (kripken) [inline emcc.py:389]: Experimental code?

File: emcc.py [hunk 2]
```diff
@@ -369,12 +369,12 @@ def get_clang_flags(user_args):
   if settings.RELOCATABLE and '-fPIC' not in user_args:
     flags.append('-fPIC')
 
-  # We use default visiibilty=default in emscripten even though the upstream
-  # backend defaults visibility=hidden.  This matched the expectations of C/C++
-  # code in the wild which expects undecorated symbols to be exported to other
-  # DSO's by default.
-  if not any(a.startswith('-fvisibility') for a in user_args):
-    flags.append('-fvisibility=default')
+    # We use default visiibilty=default in emscripten even though the upstream
```
Comment (kripken) [inline emcc.py:372]:
```suggestion
    # For relocatable code we default to visibility=default in emscripten even though the upstream
```

File: emcc.py [hunk 3]
```diff
@@ -369,12 +369,12 @@ def get_clang_flags(user_args):
...
+    # backend defaults visibility=hidden.  This matched the expectations of C/C++
```
Comment (kripken) [inline emcc.py:373]:
```suggestion
    # backend defaults visibility=hidden.  This matches the expectations of C/C++
```

**Review by kripken — APPROVED** [review]

File: emcc.py [hunk 4]
```diff
@@ -369,12 +369,12 @@ def get_clang_flags(user_args):
...
-  # code in the wild which expects undecorated symbols to be exported to other
-  # DSO's by default.
-  if not any(a.startswith('-fvisibility') for a in user_args):
-    flags.append('-fvisibility=default')
+    # For relocatable code we default to visibility=default in emscripten even
+    # though the upstream backend defaults visibility=hidden.  This matches the
+    # expectations of C/C++ code in the wild which expects undecorated symbols
+    # to be exported to other DSO's by default.
+    if not any(a.startswith('-fvisibility') for a in user_args):
+      flags.append('-fvisibility=default')
```
Comment (kripken) [inline emcc.py:377]: Re-reading this now, I realize we don't get here if `-fPIC` was passed, as the condition on line 369 checks that (and not just `RELOCATABLE`). Is that ok?

Comment (sbc100) [reply emcc.py:377]: Oops, yes this should be in `if settings.RELOCATABLE`

Comment (sbc100) [reply emcc.py:378]: Done

**Review by kripken — APPROVED** [review]

**Pull request merged**